Modules/Loadable/Segmentations/Testing/Python/SegmentationWidgetsTest1.py

```diff
@@ -77,3 +76,216 @@ def TestSection_1_qMRMLSegmentsTableView(self):
+    translationTransformNode.SetName('TestTranslation')
+    slicer.mrmlScene.AddNode(translationTransformNode)
+    translationTransformNode.SetMatrixTransformToParent(translationTransformMatrix)
+
+    tinyVolumeNode.SetAndObserveTransformNodeID(translationTransformNode.GetID())
+    geometryWidget.geometryImageData(geometryImageData)
+    self.assertTrue(self.compareOutputGeometry(geometryImageData,
+        (49,49,23), (224.3439, 223.7890, -135.25),
+        [[0.0, 0.0, 1.0], [0.0, 0.0, 1.0], [0.0, 0.0, 1.0]]))
+    slicer.util.delayDisplay('Transformed volume source - OK')
+
+    # Volume source with isotropic spacing
+    tinyVolumeNode.SetAndObserveTransformNodeID(None)
+    geometryWidget.setIsotropicSpacing(True)
+    geometryWidget.geometryImageData(geometryImageData)
+    self.assertTrue(self.compareOutputGeometry(geometryImageData,
+        (23,23,23), (248.8439, 248.2890, -123.75),
+        [[0.0, 0.0, 1.0], [0.0, 0.0, 1.0], [0.0, 0.0, 1.0]]))
+    slicer.util.delayDisplay('Volume source with isotropic spacing - OK')
+
+    # Volume source with oversampling
+    geometryWidget.setIsotropicSpacing(False)
+    geometryWidget.setOversamplingFactor(2.0)
+    geometryWidget.geometryImageData(geometryImageData)
+    self.assertTrue(self.compareOutputGeometry(geometryImageData,
+        (24.5, 24.5, 11.5), (261.0939, 260.5390, -129.5),
+        [[0.0, 0.0, 1.0], [0.0, 0.0, 1.0], [0.0, 0.0, 1.0]]))
+    slicer.util.delayDisplay('Volume source with oversampling - OK')
+
+    # Segmentation source with binary labelmap master
+    geometryWidget.setOversamplingFactor(1.0)
+    geometryWidget.setSourceNode(tinySegmentationNode)
+    geometryWidget.geometryImageData(geometryImageData)
+    self.assertTrue(self.compareOutputGeometry(geometryImageData,
+        (49,49,23), (248.8439, 248.2890, -123.75),
+        [[0.0, 0.0, 1.0], [0.0, 0.0, 1.0], [0.0, 0.0, 1.0]]))
+    slicer.util.delayDisplay('Segmentation source with binary labelmap master - OK')
+
+    # Segmentation source with closed surface master
+    tinySegmentationNode.GetSegmentation().SetConversionParameter('Smoothing factor', '0.0')
+    self.assertTrue(tinySegmentationNode.GetSegmentation().CreateRepresentation(closedSurfaceReprName))
+    tinySegmentationNode.GetSegmentation().SetMasterRepresentationName(closedSurfaceReprName)
+    tinySegmentationNode.Modified() # Trigger re-calculation of geometry (only generic Modified event is observed)
+    geometryWidget.geometryImageData(geometryImageData)
+    # Manually resample and check if the output is non-empty
+    # Note: This is done for all poly data based geometry calculations below
+    #TODO: Come up with proper geometry baselines for these too
+    vtkSegmentationCore.vtkOrientedImageDataResample.ResampleOrientedImageToReferenceOrientedImage(
+      segmentOrientedImageData, geometryImageData, geometryImageData, False, True)
+    self.assertTrue(self.imageDataContainsData(geometryImageData))
```

Comment:
Would it be possible to compare the number of non-empty voxels? It would catch much more potential bugs for sure, but of course it is only usable if it is not too fragile.

Reply:
OK. This should be easy to add.